libs/ngx-nuts-and-bolts/src/lib/table-state/table-state.helper.ts

```diff
@@ -0,0 +1,153 @@
+import { ActivatedRoute, Router } from '@angular/router';
+import { isEqual } from 'lodash';
+import { Observable } from 'rxjs';
+import { distinctUntilChanged, map } from 'rxjs/operators';
+
+export type Sort = 'asc' | 'desc' | '';
+
+export interface IPageInfo {
+	pageIndex: number;
+	pageSize: number;
+	length?: number;
+}
+
+export interface ISortInfo {
+	sortingKey: string;
+	sortDirection: Sort;
+}
+
+export enum TableQueryParam {
+	PAGE_INDEX = 'pageIndex',
+	PAGE_SIZE = 'pageSize',
+	SORT_KEY = 'sortKey',
+	SORT_DIRECTION = 'sortDirection',
+	CUSTOM_FILTERS = 'filters',
+}
+
+export function createPaginationObservable(route: ActivatedRoute): Observable<IPageInfo> {
```

Comment:
make queryParamMap the argument, add in argument for specifying fallbacks for pageIndex, pageSize

Reply:
I will add arguments for default values, but what would be the advantage of using only queryParamMap as the function argument since this method is used for reading stuff from query params?